src/equations/jin_xin_compressible_euler_2d.jl

```diff
@@ -6,190 +6,192 @@
+end
 
 
 # Calculate 1D flux in for a single point
-@inline function flux(u, orientation::Integer, equations::JinXinCompressibleEulerEquations2D)
+# TODO: Implement 1D and 3D
+@inline function flux(u, orientation::Integer, equations::JinXinEquations{2})
     if orientation == 1
-      a = equations.a
-      flux = SVector(u[5],u[6],u[7],u[8],a[1]*u[1],a[2]*u[2],a[3]*u[3],a[4]*u[4],0,0,0,0)
-    else
-      b = equations.b
-      flux = SVector(u[9],u[10],u[11],u[12],0,0,0,0,b[1]*u[1],b[2]*u[2],b[3]*u[3],b[4]*u[4])
+        u_base = get_block_components(u, 1, equations)
+        fluxes = get_block_components(u, 2, equations)
+        velocities = equations.velocities[1]
+        return SVector(fluxes..., (velocities .* u_base)..., zero(u_base)...)
+    else # orientation == 2
+        u_base = get_block_components(u, 1, equations)
+        fluxes = get_block_components(u, 3, equations)
+        velocities = equations.velocities[2]
+        return SVector(fluxes..., zero(u_base)..., (velocities .* u_base)...)
     end
-    return flux
 end
 
-
+# TODO: Implement 1D and 3D
 @inline function flux_upwind(u_ll, u_rr, orientation::Integer,
-                              equations::JinXinCompressibleEulerEquations2D)
+                             equations::JinXinEquations{2})
+    u_ll_base = get_block_components(u_ll, 1, equations)
+    u_rr_base = get_block_components(u_rr, 1, equations)
+
     if orientation == 1
-      sai = equations.sqrt_a_inv
-      sa  = equations.sqrt_a
-      #dissipation = SVector(sai[1]*(u_rr[5] - u_ll[5]), sai[2]*(u_rr[6] - u_ll[6]), sai[3]*(u_rr[7] - u_ll[7]), sai[4]*(u_rr[8] - u_ll[8]), sa[1]*(u_rr[1] - u_ll[1]), sa[2]*(u_rr[2] - u_ll[2]), sa[3]*(u_rr[3] - u_ll[3]), sa[4]*(u_rr[4] - u_ll[4]),0.0,0.0,0.0,0.0) 
-      dissipation = SVector(sa[1]*(u_rr[1] - u_ll[1]), sa[2]*(u_rr[2] - u_ll[2]), sa[3]*(u_rr[3] - u_ll[3]), sa[4]*(u_rr[4] - u_ll[4]), sa[1]*(u_rr[5] - u_ll[5]), sa[2]*(u_rr[6] - u_ll[6]), sa[3]*(u_rr[7] - u_ll[7]), sa[4]*(u_rr[8] - u_ll[8]),0.0,0.0,0.0,0.0) 
-    else
-      sbi = equations.sqrt_b_inv
-      sb  = equations.sqrt_b
-      #dissipation = SVector(sbi[1]*(u_rr[9] - u_ll[9]), sbi[2]*(u_rr[10] - u_ll[10]), sbi[3]*(u_rr[11] - u_ll[11]), sbi[4]*(u_rr[12] - u_ll[12]),0.0,0.0,0.0,0.0, sb[1]*(u_rr[1] - u_ll[1]), sb[2]*(u_rr[2] - u_ll[2]), sb[3]*(u_rr[3] - u_ll[3]), sb[4]*(u_rr[4] - u_ll[4])) 
-      dissipation = SVector(sb[1]*(u_rr[1] - u_ll[1]), sb[2]*(u_rr[2] - u_ll[2]), sb[3]*(u_rr[3] - u_ll[3]), sb[4]*(u_rr[4] - u_ll[4]),0.0,0.0,0.0,0.0,sb[1]*(u_rr[9] - u_ll[9]), sb[2]*(u_rr[10] - u_ll[10]), sb[3]*(u_rr[11] - u_ll[11]), sb[4]*(u_rr[12] - u_ll[12]))
+        sqrt_velocities = equations.sqrt_velocities[1]
+        f_ll_base = get_block_components(u_ll, 2, equations)
+        f_rr_base = get_block_components(u_rr, 2, equations)
+        dissipation = SVector((sqrt_velocities .* (u_rr_base - u_ll_base))...,
+                              (sqrt_velocities .* (f_rr_base + f_ll_base))...,
```

Comment:
@ranocha Are you sure this is supposed to be a `+` and not a `-`, as it was in @gregorgassner's original implementation?

Reply:
Should be `-` - sorry, my fault 🙈